analyzers/src/SonarAnalyzer.CSharp/Common/Walkers/CatchLoggingInvocationWalker.cs

```diff
@@ -0,0 +1,138 @@
+    public List<InvocationExpressionSyntax> LoggingInvocationsWithoutException { get; } = new();
+
+    private static readonly ImmutableArray<LoggingInvocationDescriptor> LoggingInvocationDescriptors = ImmutableArray.Create(
+        new LoggingInvocationDescriptor(MicrosoftExtensionsLogging, KnownType.Microsoft_Extensions_Logging_LoggerExtensions, false),
+        new LoggingInvocationDescriptor(CastleCoreOrCommonCore, KnownType.Castle_Core_Logging_ILogger, true),
+        new LoggingInvocationDescriptor(CastleCoreOrCommonCore, KnownType.Common_Logging_ILog, true),
+        new LoggingInvocationDescriptor(Log4NetILog, KnownType.log4net_ILog, true),
+        new LoggingInvocationDescriptor(Log4NetILogExtensions, KnownType.log4net_Util_ILogExtensions, false),
+        new LoggingInvocationDescriptor(NLogLoggingMethods, KnownType.NLog_ILogger, true),
+        new LoggingInvocationDescriptor(NLogLoggingMethods, KnownType.NLog_ILoggerExtensions, false),
+        new LoggingInvocationDescriptor(NLogILoggerBase, KnownType.NLog_ILoggerBase, true),
+        new LoggingInvocationDescriptor(Serilog, KnownType.Serilog_ILogger, true),
+        new LoggingInvocationDescriptor(Serilog, KnownType.Serilog_Log, false));
+
+    public CatchLoggingInvocationWalker(SemanticModel model)
+    {
+        this.model = model;
+    }
+
+    public override void VisitCatchClause(CatchClauseSyntax node)
+    {
+        // We want to look for logging invocations only in the main catch clause.
+        if (isFirstCatchClauseVisited)
+        {
+            return;
+        }
+
+        isFirstCatchClauseVisited = true;
+        hasWhenFilterWithDeclarations = node.Filter != null && node.Filter.DescendantNodes().Any(DeclarationPatternSyntaxWrapper.IsInstance);
+        if (node.Declaration != null && !node.Declaration.Identifier.IsKind(SyntaxKind.None))
+        {
+            caughtException = model.GetDeclaredSymbol(node.Declaration);
+        }
+        base.VisitCatchClause(node);
+    }
+
+    public override void VisitInvocationExpression(InvocationExpressionSyntax node)
+    {
+        if (!IsExceptionLogged && IsLoggingInvocation(node, model))
+        {
+            if (GetArgumentSymbolDerivedFromException(node, model) is { } currentException
+                && (hasWhenFilterWithDeclarations || currentException.Equals(caughtException)))
+            {
+                IsExceptionLogged = true;
+                LoggingInvocationWithException = node;
+                return;
+            }
+            else
+            {
+                LoggingInvocationsWithoutException.Add(node);
+            }
+        }
+        base.VisitInvocationExpression(node);
+    }
+
+    public override void VisitThrowStatement(ThrowStatementSyntax node)
+    {
+        if (ThrowStatementSyntax == null
+            && RethrowsCaughtException(node))
+        {
+            ThrowStatementSyntax = node;
+        }
+        base.VisitThrowStatement(node);
```

Comment:
Is there a reason to keep visiting the `ThrowStatement` children here?
I don't think it can contain `Invocation`, `CatchClause` or another `Throw`.